### ISSUE TITLE
nginx rewrite for rest api

### ISSUE THREAD

**Issue by grossherr** [phpList/phplist3#552]
I'm using phplist with nginx it's working so far, my configuration file, only relevant parts included, looks like this:

```
  location ~* \.(txt|log|inc)$ {
    allow 127.0.0.1;
    deny all;
  }

  location ~* \.(js|css|png|jpg|jpeg|gif|ico)$ {
    expires max;
    log_not_found off;
  }

  location /config {
    deny all;
  }

  location ~* (index\.php|upload\.php|connector\.php|dl\.php|ut\.php|lt\.php|download\.php|api\.php)$ {
    fastcgi_pass unix:/run/php/php7.2-fpm.sock;
    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    include fastcgi_params;
    include snippets/fastcgi-php.conf;
  }
```

Basically following: https://www.nginx.com/resources/wiki/start/topics/recipes/phplist/?highlight=phplist

Now I activated the rest api, following this: https://www.phplist.org/manual/ch049_api-and-integrations.xhtml

I tried several options in nginx for rewriting, e.g. something like: 

```
  location ~ ^/api/v2/ {
    rewrite ^/api/v2/(.*?)$ /api.php$is_args$1 last;
  }
```

I'm not an expert though and seem to miss something. Your help is much appreciated. Has anyone done this, and if so, how can it be done?

**Comment by michield** [comment]
I think it's probably better to ask this in an nginx forum.

But https://github.com/phpList/base-distribution#phplist-4-base-distribution
maps to app.php, not api.php 

Maybe that's the issue? Otherwise @xh3n1 or @oliverklee may have an idea?

**Comment by oliverklee** [comment]
Yes, the rewrite needs to go to `app.php`. There is no `api.php` in phpList 4.

**Comment by grossherr** [comment]
I have phpList 3.4.2 installed. There is no `app.php`, but an `api.php`. In the `.htaccess` the rewrite goes to the `api.php`, which is why I used it.

**Comment by xh3n1** [comment]
@grossherr you are right as you right as I did changes to make rest API work with phpList3.

Regarding your issue: 
I think you should add this line as it was posted in the forum https://discuss.phplist.org/t/rest-api-with-nginx/5282/15:
`rewrite ^(/lists/api/v2/.*) /lists/api.php;`
Also, the best place to ask for help and support is the phpList forum ;)

**Comment by grossherr** [comment]
Thanks to all of you. I'll give it a try, and next time turn to the forum.
